src/app/[lang]/courses/new-course/search-course-form/page.tsx

```diff
@@ -9,51 +14,48 @@ type Props = {
+            ...prevState,
+            [name]: value,
+        }));
+        localStorage.setItem(name, value);
+    }
+
+    useEffect(() => {
+        localStorage.setItem('yourStateKey', JSON.stringify(state));
+    }, [state]);
+
     return (
         <div className='flex flex-col justify-center items-center gap-4'>
             <Text className='text-xl'>שׁם הקורס</Text>
             <form className='flex flex-col items-center justify-center gap-4'>
-                <label className='text-l flex flex-row gap-3'>
-                    <Text> גיל: </Text>
-                    <input
-                        className='w-[200px] h-[30px] bg-blue-500 debug flex items-center justify-center'
-                        type='text'
-                        placeholder=''
-                        name='age'
-                    />
-                </label>
-                <label className='text-l flex flex-row gap-3'>
-                    <Text> רמת נגינה רצויה: </Text>
-                    <select
-                        id='experience'
-                        name='experience'
-                        className='block w-full p-2 border border-gray-300 rounded-md'>
-                        <option value='beginner'>Beginner</option>
-                        <option value='intermediate'>Intermediate</option>
-                        <option value='advanced'>Advanced</option>
-                    </select>
-                </label>
+                <FormInput name={'age'} onChange={handleChange} />
+                <FormSelect
+                    name={'diffcultyLevel'}
+                    option={['Beginner', 'Intermediate', 'Advanced']}
+                    onChange={handleChange}
+                />
+                <FormSelect
+                    name={'musicStyle'}
+                    option={['Jazz', 'Rock', 'House']}
+                    onChange={handleChange}
+                />
 
-                <label className='text-l flex flex-row gap-3'>
-                    <Text> סגנונות מוזיקליים: </Text>
-                    <select
-                        id='experience'
-                        name='experience'
-                        className='block w-full p-2 border border-gray-300 rounded-md'>
-                        <option value='beginner'>Jazz</option>
-                        <option value='intermediate'>Rock</option>
-                        <option value='advanced'>House</option>
-                    </select>
-                </label>
-                <label className='text-l flex flex-row gap-3'>
-                    <Text> מידת ניסיון קודם: </Text>
-                    <input
-                        className='w-[200px] h-[30px] bg-blue-500 debug flex items-center justify-center'
-                        type='text'
-                        placeholder=''
-                        name='age'
-                    />
-                </label>
+                <FormInput name={'experience'} onChange={handleChange} />
```

Comment:
In my opinion this should have been ```FormSelect``` or better explained, free text for experience is kind of unclear and confusing 
also ```age```  is a bit strange to me since this is course creation

Reply:
About the 'age,' the idea is to be able to search for courses for 5-year-old kids, for example, separately from adults.